### PR TITLE
Smarsh Connector

### LINKED ISSUE #2032: Save/log all messages for SOX/SEC compliance etc
Is there any way to keep a running backup of messages or log all the messages out to a file to be sent to a service like smarsh. We need to keep archives of all messages in some kind of restorable/searchable form for compliance as do many businesses in financial or health care. Has anyone done this or know how to approach this with rocket.chat.

### PR DESCRIPTION
**Pull request by graywolf336**
@RocketChat/core 

Closes #2032 (and maybe #2869)

Here is the first version of the Smarsh connector (might be able to rename it, if other systems can use this format). When configured, it will send emails to the configured address at the configured interval (defaults to every 30 minutes) of all of the messages which happen in all of the rooms including direct messages. File uploads in a room will only be attached to the email if the uploads are stored in the file system or the database.

screen shot 2016-08-27 at 9 07 41 pm

### CODE REVIEW

**Comment by dupree972** [comment]
The other main compliance retention product GlobalRelay also accepts in the same manner,  I would think Iron mountain would be similar but am not sure
Bill

File: packages/rocketchat-lib/i18n/en.i18n.json [hunk 1]
```diff
@@ -1067,6 +1070,12 @@
   "Slash_TableUnflip_Description" : "Displays ┬─┬﻿ ノ( ゜-゜ノ)",
   "Slash_Topic_Description" : "Set topic",
   "Slash_Topic_Params" : "Topic message",
+  "Smarsh_Enabled": "Smarsh Enabled",
+  "Smarsh_Enabled_Description": "Whether the Smarsh eml connector is enabled or not (needs 'From Email' filled in under Email -> SMTP).",
+  "Smarsh_Email": "Smarsh Email",
+  "Smarsh_Email_Description": "Smarsh Email Address to send the .eml file to.",
+  "Smarsh_Interval": "Smarsh Internval",
```
Comment (geekgonecrazy) [inline packages/rocketchat-lib/i18n/en.i18n.json]: Small typo :) Should be:
> Smarsh Interval

File: packages/rocketchat-smarsh-connector/server/settings.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	this.add('Smarsh_Email', '', {
+		type: 'string',
+		i18nLabel: 'Smarsh_Email',
+		placeholder: 'email@domain.com'
+	});
+	this.add('Smarsh_Interval', 'every_30_minutes', {
+		type: 'select',
+		values: [{
+			key: 'every_30_seconds',
+			i18nLabel: 'every_30_seconds'
```
Comment (graywolf336) [inline packages/rocketchat-smarsh-connector/server/settings.js]: This 30 seconds one is more for testing purposes and not really for meant for production (so we don't have to wait a while), so before we release this to the `master` branch we will want to remove this option.

**Comment by graywolf336** [comment]
@RocketChat/core Any chance I can get some eyes on this for opinions and maybe even some looks good so we can get it in `develop` builds?

**Comment by sampaiodiego** [comment]
LGTM

My only concern is with `_getLink` function.. I **think** `attachments.title_link` can be a full URI

**Comment by graywolf336** [comment]
@sampaiodiego That's what the `url.match(/^(https?:)?\/\//i)` check is for, if it does contain a `http(s)` then it will just return that url instead of appended the site's url.

**Comment by sampaiodiego** [comment]
right! I missed that :relaxed:

**Comment by kuzhao** [comment]
@graywolf336 Do we have this feature included in the latest release? Did not find any doc about this, nor the counterpart in 0.56 configurations.

**Comment by geekgonecrazy** [comment]
@kuzhao this feature is definetely in 0.56.0 its been in a few releases now

image
As to documentation.. I don't know that its documented.  But as long as you have email configured, it should be pretty easy to get going

**Comment by RYG81** [comment]
IS this feature available on self-hosted version?

**Comment by graywolf336** [comment]
@RYG81 yes, it is available in all versions

**Comment by RYG81** [comment]
Hi,

Not sure but I am not able to enable it.
Here is the attached image.

**Comment by RYG81** [comment]
Hi,

Not sure but I am not able to enable it.
Here is the attached image.

**Comment by graywolf336** [comment]
Is the `From_Email` filled in underneath the STMP settings?

**Comment by RYG81** [comment]
After adding an email address I am able to enable it. However, I am not receiving any email when I press the send the test email button right below the from email field.

**Pull request merged**